lisa/tools/dmesg.py

```diff
@@ -88,6 +86,6 @@ def _run(self, force_run: bool = False) -> ExecutableResult:
         result = self.run(force_run=force_run, no_error_log=True)
         if result.exit_code != 0:
             # may need sudo
-            result = self.run(sudo=True, force_run=force_run)
+            result = self.run(sudo=True, force_run=force_run, expected_exit_code=0)
```

Comment:
Remove the extra assert code.

Reply:
Moved assert from check_kernel_errors
https://github.com/microsoft/lisa/pull/2196/files#r944091083